src/AutoBalance.cpp

```diff
@@ -303,6 +385,23 @@ class AutoBalance_PlayerScript : public PlayerScript
                 }
             }
         }
+
+        void OnBeforeLootMoney(Player* player, Loot* loot) override
+        {
+            if (DungeonScaleDownMoney)
+            {
+                Map* map = player->GetMap();
+
+                if (map->IsDungeon())
+                {
+                    // Ensure that the players always get the same XP, even when entering the dungeon alone
+                    uint32 maxPlayerCount = ((InstanceMap*)sMapMgr->FindMap(map->GetId(), map->GetInstanceId()))->GetMaxPlayers();
+                    uint32 currentPlayerCount = map->GetPlayersCountExceptGMs();
```

Comment:
use `auto` as type when you can

Reply:
The rest of the module and AzerothCore generally does not use auto. Currently it is consistent with the rest of the module, but I can change to use auto if you would prefer. Would you like me to change these to auto?

Reply:
yes, it is better
usually we try to use the best practices when we can, some piece of code of AC are surely using `auto` but not all of them unfortunately

Reply:
Okay, I will make the change.